src/ZipStream.php

```diff
@@ -251,297 +330,241 @@ public function __construct($name = null, $opt = array())
      * @throws \ZipStream\Exception\FileNotFoundException
      * @throws \ZipStream\Exception\FileNotReadableException
      */
-    public function addFileFromPath($name, $path, $opt = array(), $storage_method = "deflate")
+    public function addFileFromPath($name, $path, $opt = array(), $method = 'deflate')
     {
-        if (!is_readable($path)) {
-            if (!file_exists($path)) {
-                throw new FileNotFoundException($path);
-            }
-            throw new FileNotReadableException($path);
-        }
-
-        if ($this->isLargeFile($path)) {
-            // file is too large to be read into memory; add progressively
-            $this->addLargeFile($name, $path, $opt);
-        } else {
-            // file is small enough to read into memory; read file contents and
-            // handle with addFile()
-
-            $data = file_get_contents($path);
-            $this->addFile($name, $data, $opt, $storage_method);
-        }
+        $file = new File($this, $name, $opt, $method);
+        $file->processPath($path);
     }
 
     /**
-     * Is this file larger than large_file_size?
+     * addFile_from_stream
      *
-     * @param string $path
-     * @return Boolean
+     * dds an open stream to the archive uncompressed
```

Comment:
`s/dds/adds/`

Reply:
Fixed. *(This typo came from your repo.)*